### PR TITLE
Remove double TryUpdateModelAsync section from Model Binding.

### LINKED ISSUE #16262: TryUpdateModelAsync described twice in Model Binding
Whilst working on #16240, I noticed that a change I made yesterday means `TryUpdateModelAsync` is now covered twice. I think the new section "Prevent over posting with TryUpdateModelAsync" should be dropped and the new link should be added to "Manual model binding".

@Rick-Anderson: What do you think? I'll PR if you agree.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: e24f48f8-6c4d-ecda-6ce6-a7826d396e99
* Version Independent ID: 9c4af557-8d06-4c6c-ad2a-0dd234547563
* Content: [Model Binding in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/mvc/models/model-binding?view=aspnetcore-3.1#manual-model-binding)
* Content Source: [aspnetcore/mvc/models/model-binding.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/mvc/models/model-binding.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-mvc**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by serpent5**
Fixes #16262.

### CODE REVIEW

**Comment by scottaddie** [comment]
I think we should actually leave this section. It provides some good info that isn't covered in the EF Core with Razor Pages tutorial. I also think it's two different audiences we're targeting with the 2 docs. Tutorials are geared more for the beginner. Thoughts?

**Comment by Rick-Anderson** [comment]
Additionally it's got great SEO now. 
I was thinking you'd combine the sections. 

we also need  `Not used with Web API`

Maybe it's best to leave them as they are now.

**Comment by Rick-Anderson** [comment]
@scottaddie  @serpent5  check my version, which combines the two sections

**Comment by serpent5** [comment]
I don't think the headline for `TryUpdateModelAsync` is for protecting against overposting. I think it's about re-running the model binding process or just doing it imperatively.

> we also need `Not used with Web API`

I don't really know what is meant by "Web API". If this means "endpoints that take JSON", that's more about not being compatible with input formatters. There's no reason a "Web API" that consumes form data wouldn't use `TryUpdateModelAsync`.

That's just my thinking. You guys are in charge here so I'm happy to go with whatever you decide.

**Comment by Rick-Anderson** [comment]
@serpent5  good points. Can you tweak my update or suggest changes?

File: aspnetcore/mvc/models/model-binding.md [hunk 1]
```diff
@@ -464,12 +455,19 @@ To disable validation on properties of a specified type, add a <xref:Microsoft.A
...
+## Manual model binding	
 
 Model binding can be invoked manually by using the <xref:Microsoft.AspNetCore.Mvc.ControllerBase.TryUpdateModelAsync*> method. The method is defined on both `ControllerBase` and `PageModel` classes. Method overloads let you specify the prefix and value provider to use. The method returns `false` if model binding fails. Here's an example:
 
 [!code-csharp[](model-binding/samples/3.x/ModelBindingSample/Pages/InstructorsWithCollection/Create.cshtml.cs?name=snippet_TryUpdate&highlight=1-4)]
 
+<xref:Microsoft.AspNetCore.Mvc.ControllerBase.TryUpdateModelAsync*>  uses value providers to get data from the form body, query string, and route data. `TryUpdateModelAsync` is typically:	
+
+* Used with Razor Pages and MVC apps using controllers and views to prevent over-posting.
+* Not used with Web API unless consumed from form data, query strings, and route data. Web API endpoints that consume JSON use [Input formatters](#input-formatters) to deserialize the request body into an object.
```
Comment (scottaddie) [inline aspnetcore/mvc/models/model-binding.md]:
```suggestion
* Not used with a web API unless consumed from form data, query strings, and route data. Web API endpoints that consume JSON use [Input formatters](#input-formatters) to deserialize the request body into an object.
```

**Comment by serpent5** [comment]
The latest changes look fine to me.

**Comment by Rick-Anderson** [comment]
@serpent5  thanks for the PR.

**Comment by serpent5** [comment]
No worries. I don't think I deserved a commit for this in the end. 😆

**Comment by Rick-Anderson** [comment]
I do. You're suggestions were a big improvement.

**Pull request merged**